docs/docs/supreme.md

```diff
@@ -621,6 +621,25 @@ The second option, however, will only allow passing a nonce/IV if the algorithm
 has the corresponding characteristic.
 The same holds for the auth tag and additional authenticated data.
 
+## Key Derivation / Key Stretching
+
+The Supreme KMP crypto provider implements the following key derivation functions:
+
+* _HKDF_ as per [RFC 5869](https://tools.ietf.org/html/rfc5869)
+* _PNKDF2_ in accordance with [RFC 8018](https://datatracker.ietf.org/doc/html/rfc8018)
+* _scrpyt_ as defined by [Colin Percival for the _Tarsnap_ online backup service](https://www.tarsnap.com/scrypt.html)
+
+Usage is the same across implementations:
+
+1. Instantiate a KDF implementation using algorithm-specific parameters as per the respective RFCs. These are:
+    * HKDF comes predefined for the SHA-1 and SHA-2 family of hash functions as `HKDF.SHA1`..`HKDF.SHA512`. Those enums require the instantiation of the nested `WithInfo(info)` class to obtain a fully instantiated object implementing the `KDF` interface.
```

Comment:
Do we have a more concise way of saying "use `HKDF.SHA1.WithInfo(foo)`"?

Reply:
i am also curious why we don't allow `HKDF.SHA1(info = foo)` directly; invoke method on the SHA1 holder returning a KDF?

Reply:
same goes for the others

Reply:
> i am also curious why we don't allow `HKDF.SHA1(info = foo)` directly; invoke method on the SHA1 holder returning a KDF?

info is mandatory. this prevents you from "forgetting" it and keeps the KDF interface the way it should be

Reply:
done